### PR TITLE
[X86] Align other variants to use void * as 512 variants.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
For *_stream_* series intrinsics

### CODE REVIEW

**Comment by FreddyLeaf** [comment]
Here's the change for 512 variants before: https://reviews.llvm.org/D66786

File: clang/test/CodeGen/X86/sse41-builtins.c [hunk 1]
```diff
@@ -358,6 +358,12 @@ __m128i test_mm_stream_load_si128(__m128i const *a) {
   return _mm_stream_load_si128(a);
 }
 
+__m128i test_mm_stream_load_si128_void(void const *a) {
```
Comment (phoebewang) [inline clang/test/CodeGen/X86/sse41-builtins.c:361]: `const void`?

File: clang/lib/Headers/smmintrin.h [hunk 2]
```diff
@@ -645,7 +645,7 @@ static __inline__ __m128i __DEFAULT_FN_ATTRS _mm_mul_epi32(__m128i __V1,
 /// \returns A 128-bit integer vector containing the data stored at the
 ///    specified memory location.
 static __inline__ __m128i __DEFAULT_FN_ATTRS
-_mm_stream_load_si128(__m128i const *__V) {
+_mm_stream_load_si128(void const *__V) {
```
Comment (phoebewang) [inline clang/lib/Headers/smmintrin.h:648]: Move `const` first?

File: clang/lib/Headers/avx2intrin.h [hunk 3]
```diff
@@ -2979,7 +2979,7 @@ _mm256_xor_si256(__m256i __a, __m256i __b)
 ///    A pointer to the 32-byte aligned memory containing the vector to load.
 /// \returns A 256-bit integer vector loaded from memory.
 static __inline__ __m256i __DEFAULT_FN_ATTRS256
-_mm256_stream_load_si256(__m256i const *__V)
+_mm256_stream_load_si256(void const *__V)
```
Comment (phoebewang) [inline clang/lib/Headers/avx2intrin.h:2982]: Move `const` first?

**Review by phoebewang — APPROVED** [review]
LGTM.

File: clang/test/CodeGen/X86/avx2-builtins.c [hunk 4]
```diff
@@ -1223,6 +1223,12 @@ __m256i test_mm256_stream_load_si256(__m256i const *a) {
   return _mm256_stream_load_si256(a);
 }
 
+__m256i test_mm256_stream_load_si256_const(void const *a) {
```
Comment (RKSimon) [inline clang/test/CodeGen/X86/avx2-builtins.c:1226]: test_mm256_stream_load_si256_void ?

Comment (FreddyLeaf) [reply clang/test/CodeGen/X86/avx2-builtins.c]: 3e463d3

File: clang/lib/Headers/avx2intrin.h (hunk 3 above)
Comment (FreddyLeaf) [reply clang/lib/Headers/avx2intrin.h]: 3e463d3c41bb8e8391c62dd95910d009b4b39b7d

File: clang/lib/Headers/smmintrin.h (hunk 2 above)
Comment (FreddyLeaf) [reply clang/lib/Headers/smmintrin.h]: 3e463d3c41bb8e8391c62dd95910d009b4b39b7d

File: clang/test/CodeGen/X86/sse41-builtins.c (hunk 1 above)
Comment (FreddyLeaf) [reply clang/test/CodeGen/X86/sse41-builtins.c]: 3e463d3c41bb8e8391c62dd95910d009b4b39b7d

**Review by RKSimon — APPROVED** [review]
LGTM - please update the description to mention that you're updating the movnti i32/i64 scalar integer nt ops as well as the vector ops

**Review by KanRobert — APPROVED** [review]
LGTM

**Comment by RKSimon** [comment]
For completeness the _mm_stream_sd / _mm_stream_ss SSE4A intrinsics still need updating as well: https://github.com/llvm/llvm-project/blob/59fbba94908f65eedb8bdd619e425bf97d84b2e3/clang/lib/Headers/ammintrin.h#L158C1-L158C14

**Comment by FreddyLeaf** [comment]
> For completeness the _mm_stream_sd / _mm_stream_ss SSE4A intrinsics still need updating as well: https://github.com/llvm/llvm-project/blob/59fbba94908f65eedb8bdd619e425bf97d84b2e3/clang/lib/Headers/ammintrin.h#L158C1-L158C14

Good catch. Seems like intrinsic guide missed these two.

**Comment by RKSimon** [comment]
They're AMD specific so probably not covered

**Comment by FreddyLeaf** [comment]
> LGTM - please update the description to mention that you're updating the movnti i32/i64 scalar integer nt ops as well as the vector ops

I think this description has covered? May display wrong in the preview, highlighted here:
`For *_stream_* series intrinsics`

**Comment by RKSimon** [comment]
I meant the patch title - I'm never quite sure how the title/description appears when these are squashed+merged

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Pull request merged**